index.d.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Wraps the AssemblyScript build in a JS function.
+ * This allows users to not have to make AS a dependency in their project.
+ * @param {Uint8Array} message Message to hash
+ */
+declare function sha256(message): Uint8Array;
+
+export declare function clean(): void;
+
+export declare function init(): void;
```

Comment:
Its a little strange that we only allow for a single hash context.
I think we'll only be using the all-at-once default function here, so it won't be a problem for us.

Reply:
I'm pretty sure in ssz we need to call update no?

Reply:
I was planning on changing our function in hash to:
```typescript
import sha256 from "@chainsafe/sha256";
export function hash(input: Buffer): Buffer {
  return Buffer.from(sha256(input));
}
```

The single hash context I'm talking about is using the hash builder api.
Imagine you want to be creating multiple sha hashes at once. Usually you would do something like:
```typescript
let h1 = sha256.init();
let h2 = sha256.init();

h1 = h1.update(foo);
h2 = h2.update(bar);

h1 = h1.final();
h2 = h2.final();
```
The "strange" thing I was referring to is that you can't do this pattern above, because the library only allows calculating a single hash at a time. I'm not familiar with the limitations of the current implementation, just thought its worth noting.

Reply:
@tueric and i will chat about this now...

Reply:
test

Reply:
@wemeetagain Wouldn't that require allocating huge array to do it in single pass. Memory wise I think ucrrent setup with multiple update is much more appropriate

Reply:
@mpetrunic practically speaking we're already doing this.
If you look through our codebase, you'll see `hash(Buffer.concat[a,b,c]))` everywhere.